app/cluster/dynamic.go

```diff
@@ -39,12 +48,14 @@ type Dynamic struct {
 
 	MultiTenantStat lifecycle
 
-	currentMode servermode.Mode
+	currentMode         servermode.Mode
+	currentWorkspaceIDs string
 
 	serverStartTimeStat  stats.RudderStats
 	serverStopTimeStat   stats.RudderStats
 	serverStartCountStat stats.RudderStats
 	serverStopCountStat  stats.RudderStats
+	BackendConfig        startStopPooling
```

Comment:
You mean why using this interface:
```go
	Stop()
	StartWithIDs(workspaces string)
	WaitForConfig(ctx context.Context) error
```

Instead of:
```
type BackendConfig interface {
	SetUp()
	Get(string) (ConfigT, bool)
	GetWorkspaceIDForWriteKey(string) string
	GetWorkspaceIDForSourceID(string) string
	GetWorkspaceLibrariesForWorkspaceID(string) LibrariesT
	WaitForConfig(ctx context.Context) error
	Subscribe(channel chan pubsub.DataEvent, topic Topic)
	Stop()
	StartWithIDs(workspaces string)
}
```